### ISSUE TITLE
PL field in VCF output

### ISSUE THREAD

**Issue by amkozlov** [biosinodx/SCcaller#10]
Hi guys,

currently, `PL` field in the VCF output of SCcaller is defined as follows:

```
##FORMAT=<ID=PL,Number=G,Type=Integer,Description="sequencing noise, amplification artifact, heterozygous SNV and homozygous SNV respectively">
```

whereas according to [VCF4.3 spec](http://samtools.github.io/hts-specs/VCFv4.3.pdf) it has a different meaning:
```
PL  (Integer):  The  phred-scaled  genotype  likelihoods  rounded  to  the  closest  integer,  and  otherwise  definedprecisely as the GL field
GL (Float):  Genotype likelihoods comprised of comma separated floating point log10-scaled likelihoods for all possible genotypes given the set of alleles defined in the REF and ALT fields.
```

Could you please consider printing genotype likelihoods ('0/0', '0/1', '1/1') into 'PL' field as demanded by the spec? This would make it much easier to parse for downstream programs which rely on the VCF spec.

You could then use another, custom field to print additional info (`sequencing noise` etc.).

Thanks!

**Comment by biosinodx** [comment]
Hi Alexey,

Thanks a lot for the kind sugguestion. We will definitely consider it in our next update, which is coming soon.

Best wishes,
Xiao

获取 Outlook for Android<https://aka.ms/ghei36>



发件人: Alexey Kozlov
发送时间: 7月13日星期六 上午12:35
主题: [biosinodx/SCcaller] PL field in VCF output (#10)
收件人: biosinodx/SCcaller
抄送: Subscribed


Hi guys,
currently, PL field in the VCF output of SCcaller is defined as follows:
##FORMAT=<ID=PL,Number=G,Type=Integer,Description="sequencing noise, amplification artifact, heterozygous SNV and homozygous SNV respectively">
whereas according to VCF4.3 spec<http://samtools.github.io/hts-specs/VCFv4.3.pdf> it has a different meaning:
PL (Integer): The phred-scaled genotype likelihoods rounded to the closest integer, and otherwise definedprecisely as the GL field GL (Float): Genotype likelihoods comprised of comma separated floating point log10-scaled likelihoods for all possible genotypes given the set of alleles defined in the REF and ALT fields.
Could you please consider printing genotype likelihoods ('0/0', '0/1', '1/1') into 'PL' field as demanded by the spec? This would make it much easier to parse for downstream programs which rely on the VCF spec.
You could then use another, custom field to print additional info (sequencing noise etc.).
Thanks!
―

**Comment by amkozlov** [comment]
great, thank you very much, Xiao!

In the meantime, is there an easy way to compute REF/REF, REF/ALT and ALT/ALT genotype likelihoods from the four values reported in `PL` field? Then we could implement a temporary workaround for our analysis.

Best,
Alexey

**Comment by biosinodx** [comment]
Yes, the first two value are for REF/REF(i suggest simply to take the bigger number as the PL combined), third for REF/ALT, last for ALT/ALT.

Xiao

?? Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by amkozlov** [comment]
perfect, thanks for your fast reply!

**Comment by jmfa** [comment]
Hi everyone, 
Small follow-up question:
When you suggest to take the "the bigger number as the PL combined" of the first 2 values as the REF/REF likelihood, do you mean the largest value of the two or the one with the "the highest likelihood" (which in this case would be the smallest integer value)?

Thanks in advance
J

**Comment by amkozlov** [comment]
So let's say we have
```
20,32,91,1546
```
should it be converted to a)
```
20,91,1546
```
or b)

```
32,91,1546
```
?

Thanks!
